x-team/xp-ui
# Add unit tests for helpers methods

**Release Type:** *Dev Improvement* 


Fixes https://github.com/x-team/xp/issues/2017

## Description

Unit tests added for the helper methods of XP-UI

## Checklist

**Before submitting a pull request,** please make sure the following is done:



## Steps to Test or Reproduce

1. Run `rm -r node_dependencies && npm i`
2. Run `npm run jest`
3. Confirm that the unit tests added for the helpers methods all pass

![Screen Shot on 2020-04-17 at 01:31:27](https://user-images.githubusercontent.com/6010411/79535358-42369f80-804b-11ea-91a4-fdd701cdca9e.png)

## Impacted Areas in Application

- `src/utils/helpers.js` - added comments and removed dead methods
1. `getOlderBrowserErrorKey()`- refactored
2. `timeSince()`- refactored
- `_tests_/utils/helpers.spec.js` - created
- `.jest/setupTests.js` - refactored

## Screenshots

1. Affected by `timeSince()` helper method refactored

- `/src/components/ui/ActivityBar.js`

![Screen Shot on 2020-04-17 at 00:57:53](https://user-images.githubusercontent.com/6010411/79533691-7bb8dc00-8046-11ea-84fc-b876585b8241.png)

- `/src/components/ui/Email.js`

![Screen Shot on 2020-04-17 at 00:58:16](https://user-images.githubusercontent.com/6010411/79533707-896e6180-8046-11ea-8fd7-692a0f02f4d8.png)

![Screen Capture on 2020-04-17 at 00-59-22](https://user-images.githubusercontent.com/6010411/79533787-c5a1c200-8046-11ea-94f8-2f3af4e49d0e.gif)

- `/src/components/ui/EmailFeed.js`

![Screen Capture on 2020-04-17 at 01-01-16](https://user-images.githubusercontent.com/6010411/79533881-0a2d5d80-8047-11ea-981f-9e31ebd58045.gif)

- `/src/components/ui/Note.js`

![Screen Shot on 2020-04-17 at 01:02:43](https://user-images.githubusercontent.com/6010411/79533942-33e68480-8047-11ea-8f7a-0de842106015.png)

![Screen Shot on 2020-04-17 at 01:03:00](https://user-images.githubusercontent.com/6010411/79533946-36e17500-8047-11ea-907d-594b151b3282.png)

2. Affected by `getOlderBrowserErrorKey()` helper method refactored

- `/src/components/ui/ErrorBox.js`

![Screen Capture on 2020-04-17 at 01-08-57](https://user-images.githubusercontent.com/6010411/79534290-24b40680-8048-11ea-82e4-e1546dfd28f5.gif)
